src/nupic/algorithms/sdr_classifier.py

```diff
@@ -369,6 +363,90 @@ def inferSingleStep(self, patternNZ, weightMatrix):
     return predictDist
 
 
+  @classmethod
+  def read(cls, proto):
+    classifier = object.__new__(cls)
+
+    classifier.steps = []
+    for step in proto.steps:
```

Comment:
Only a performance impact with large values of `proto.steps`, but repeated `list.append()`'s can be slower than alternatives like list comprehensions.  For example:

```python
classifier.steps = [step for step in proto.steps]
```

For that matter, why not `classifier.steps = proto.steps`?